reactive/kotlinx-coroutines-reactor/src/Mono.kt

```diff
@@ -89,3 +132,88 @@ public fun <T> CoroutineScope.mono(
     block: suspend CoroutineScope.() -> T?
 ): Mono<T> = monoInternal(this, context, block)
 
+/**
+ * Awaits the first value from the given publisher without blocking the thread and returns the resulting value, or, if
+ * the publisher has produced an error, throws the corresponding exception.
+ *
+ * This suspending function is cancellable.
+ * If the [Job] of the current coroutine is cancelled or completed while the suspending function is waiting, this
+ * function immediately cancels its [Subscription] and resumes with [CancellationException].
+ *
+ * @throws NoSuchElementException if the publisher does not emit any value
+ */
+@Deprecated(
+    message = "Mono produces at most one value, so the semantics of dropping the remaining elements are not useful. " +
+        "Please use awaitSingle() instead.",
+    level = DeprecationLevel.WARNING,
+    replaceWith = ReplaceWith("this.awaitSingle()")
+)
+public suspend fun <T> Mono<T>.awaitFirst(): T = awaitSingle()
```

Comment:
It's a "lint" signature, so we typically expect them to be similar to ones from `kotlinx.coroutines.flow.Migration` -- `ERROR` level, no body, `KDoc` with an explanation of our reasoning.

But this one is slightly different: it potentially shadows existing extensions from `Publisher`. So I suggest leaving it as `WARNING` with a body for now, but get rid of the current `KDoc` and only explain reasoning of deprecation.

Also, worth dropping `// For migration, added with warning in 1.5, should be error in 1.6 and never hidden` after the annotation